### DISCUSSION TITLE
“Add to dictionary” in Collabora Online on Nextcloud AIO

### DISCUSSION THREAD

**Discussion by karlemilnikka** [CollaboraOnline/online#5238] [Q&A] [answered]
Has anyone got “add to dictionary” to work in Collabora Online, running on Nextcloud AIO? The option is missing when right-clicking on words that are indicated to be misspelled.

**Comment by karlemilnikka** [comment]
Here’s what I see when right-clicking. 
<img width="877" alt="Skärmavbild 2022-09-11 kl  10 02 54" src="https://user-images.githubusercontent.com/20514810/189517860-a1d6d6c4-3759-410a-99dc-7c3dfc640fb3.png">

**Comment by thebearon** [comment, marked as answer]
It's been removed, because it can't be done at the moment, the words added wouldn't be persistent.

**Reply by karlemilnikka** [reply]
Thank you very much for your quick reply.

**Comment by karlemilnikka** [comment]
@thebearon Do you know if “ignore all” is supposed to work? When right-clicking and choosing “ignore all” nothing happens. Ignoring one mislabeled misspelling at a time works.

**Comment by bentuna** [comment]
We still have the same Problems here in version 23.0.5 running on Nextcloud 27.0.2. Is there any news on the topic?

**Comment by Darkdragon84** [comment]
@thebearon when do you reckon it can be done? What's missing?

**Reply by thebearon** [reply]
These words should be saved in the integrating environment (eg. Nextcloud), the feature would need to implemented on that end, and as an interface to Collabora Online. There are also tricky situations, because originally, in LibreOffice these additions are saved to the user profile, but in this case not only a single user can work on a document, so storing these per user on the file host might not make much sense.

**Comment by P4z** [comment]
@thebearon Could you tell us why the words added wouldn't be persistent and is it possible to do this, please?

**Reply by thebearon** [reply]
Collabora Online is stateless, once a session is finished and the changes are saved, the session is discarded, including the personal dictionary items that were added during the session. The feature comes from LibreOffice code, and the personal dictionary is saved into the user profile, which doesn't exist, and doesn't make sense in Collabora Online.

**Reply by P4z** [reply]
Thank you for the explanation.

**Comment by asandikci** [comment]
@thebearon what is the current status for this feature? Are there any way to implement this feature that we should prefer? What are the roadmap and requirements for this feature? Should we open a new issue for this?

---

My thoughts about implementation:
- there should we 2 dictionary type (user-defined / server-admin-defined)
- these dictionaries should be stored either on collabora-server`*` or on a admin-specified nextcloud folder
- user should have "add to dictionary" option when right clicking an unknown word. This option should be also in "Spelling" section.
- server admins`**` should have "add to dictionary" and "add to global dictionary" options when right click menu/Spelling section.
- when a document opened, both dictionaries (user-defined dictionary of current user / server-admin-defined) should be loaded. But this actually requires to store all user names... I see why this not that easy while typing this lines lol
- Maybe we don't need to store all user names. If dictionaries will be store in Nextcloud side `user-defined` dictionaries could be located in ".office/user-dict.dct?" for example. This path should be also changeable in Nextcloud > Administration Settings > Office > Advanced Settings. But is it possible to pass a file to collabora server from nextcloud apart from current odf document? Let me know!

`*`this possibly requires user database integration between collabora/nextcloud and maybe will be more complicated 
`**`how to recognize if the user is server admin or not? Maybe add a new setting in collabora admin.html page for entering admin username. And/or and access token for more security.

---

If there is anything that I can help with idea (and maybe coding) please let me know. Also I want to monetize this feature request with "10$" if it will be implemented. (I'm student and thats the only amount of I can donate)

English is not my native language, so sorry if there was any mistake

**Reply by jacobgkau** [reply]
> **how to recognize if the user is server admin or not? Maybe add a new setting in collabora admin.html page for entering admin username. And/or and access token for more security.

Over at https://github.com/CollaboraOnline/online/issues/10092#issuecomment-2415796561, @eszkadev indicated that Collabora can already tell if a user is a Nextcloud admin or not (and uses that information to decide whether to display certain warnings), so adding a new mechanism for that may not be necessary.
